kombu/connection.py

```diff
@@ -815,8 +815,18 @@ def default_channel(self):
             a connection is passed instead of a channel, to functions that
             require a channel.
         """
+        ensure_connection_options = {}
+        if self.transport_options:
+            options_keys = [
+                'max_retries',
+                'interval_start', 'interval_step', 'interval_max',
+            ]
+            for k in options_keys:
```

Comment:
Can you help me understand your reasoning for the aversion to using a loop?

I would recommend against using `transport_options.get('foo', None)`, being that the default response if no key is provided will override the defaults provided in the `ensure_connection` signature:

```python

    def ensure_connection(self, errback=None, max_retries=None,
                          interval_start=2, interval_step=2, interval_max=30,
                          callback=None, reraise_as_library_errors=True):
```

We could first check if the key is present in the options, however I feel that this is a bit repetitive. However, if you're against the loop for the sake of readability, then this is probably the best way to go.

```python
if 'max_retries' in self.transport_options:
    opts['max_retries'] = self.transport_options['max_retries']
if 'interval_start' in self.transport_options:
    opts['interval_start'] = self.transport_options['interval_start']
if 'interval_step' in self.transport_options:
    opts['interval_step'] = self.transport_options['interval_step']
if 'interval_max' in self.transport_options:
    opts['interval_max'] = self.transport_options['interval_max']
```